lib/perl/OVH/Bastion/allowdeny.inc

```diff
@@ -209,6 +231,59 @@ sub is_access_way_granted {
             }
         }
 
+        # make sure both proxyIp and proxyPort are defined or undefined
+        if (defined $wantedProxyIp && !defined $wantedProxyPort) {
+            return R('ERR_INVALID_PARAMETER', msg => "If proxyIp is given, proxyPort must be given too");
+        }
```

Comment:
As these vars don't depend on what we're doing in the loop, we can move this block above the loop so it's checked once and for all